crates/red_knot_python_semantic/src/types/call/arguments.rs

```diff
@@ -1,88 +1,179 @@
+use std::cell::Cell;
+use std::rc::Rc;
+
+use bitflags::bitflags;
+
 use super::Type;
 
-/// Typed arguments for a single call, in source order.
+/// Typed arguments for a single call, in source order, with an optional bound self/cls parameter.
```

Comment:
I think it would be good to document that `CallArguments` should be light to clone because the operation is performent frequently during call binding.

Reply:
We don't clone this anymore, we use the `push_self`/`pop_self` methods to modify a single one in place.